File: 645-set-mismatch/set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {

        int n = nums.length;

        int[] freq = new int[n + 1];

        for (int num : nums) {
            freq[num]++;
        }

        int duplicate = -1;
        int missing = -1;

        for (int i = 1; i <= n; i++) {

            if (freq[i] == 2) {
                duplicate = i;
            }

            if (freq[i] == 0) {
                missing = i;
            }
        }

        return new int[]{duplicate, missing};
    }
}

File: 645-set-mismatch/set-mismatch.py
class Solution:
    def findErrorNums(self, nums):
        n = len(nums)

        freq = [0] * (n + 1)

        for num in nums:
            freq[num] += 1

        duplicate = -1
        missing = -1

        for i in range(1, n + 1):

            if freq[i] == 2:
                duplicate = i

            if freq[i] == 0:
                missing = i

        return [duplicate, missing]